Simplenote/src/main/java/com/automattic/simplenote/utils/AccountVerificationWatcher.java

```diff
@@ -29,33 +27,44 @@
     }
 
     private final Simplenote simplenote;
-    private final List<VerificationStateListener> listeners = new ArrayList<>();
+    private final VerificationStateListener listener;
+    private Status currentState = Status.UNKNOWN;
 
-    public AccountVerificationWatcher(Simplenote simplenote) {
+    public AccountVerificationWatcher(Simplenote simplenote, VerificationStateListener listener) {
         this.simplenote = simplenote;
+        this.listener = listener;
     }
 
-    public void addVerificationStateListener(VerificationStateListener listener) {
-        listeners.add(listener);
+    private void notifyListener(Status status) {
+        if (listener != null) {
+            listener.onUpdate(status);
+        }
     }
 
-    private void notifyListeners(Status status) {
-        for (VerificationStateListener listener : listeners) {
-            listener.onUpdate(status);
+    private void updateState(Status newState) {
+        if (newState != currentState) {
+            currentState = newState;
+            notifyListener(newState);
         }
     }
 
+    private boolean isValidChangeType(Bucket.ChangeType type, String key) {
+        return type == Bucket.ChangeType.INDEX ||
+                (type == Bucket.ChangeType.INSERT && KEY_EMAIL_VERIFICATION.equals(key)) ||
+                (type == Bucket.ChangeType.MODIFY && KEY_EMAIL_VERIFICATION.equals(key));
```

Comment:
this is fine, but it's also reducible. we know that if we have a matching `key` then it means that the verification object was involved in the change. if not, the only bucket-wide change we care about is `INDEX`

```java
return KEY_EMAIL_VERIFICATION.equals(key) || type == Bucket.ChangeType.INDEX;
```

not important, just wanted to point it out. yeah I know, this also treats `REMOVE` as impacting `verification`.